core/src/main/scala/sttp/client/ResponseAs.scala

```diff
@@ -142,8 +142,7 @@ object ResponseAs {
       }
 }
 
-sealed abstract class ResponseError[+T] extends Exception {
-  def body: String
-}
-case class HttpError(body: String) extends ResponseError[Nothing]
-case class DeserializationError[T](body: String, error: T) extends ResponseError[T]
+sealed abstract class ResponseErrorTyped[+HE, +DE] extends Exception
+case class HttpError[HE](body: HE) extends ResponseErrorTyped[HE, Nothing]
+case class UnmatchedHttpError(body: String) extends ResponseErrorTyped[Nothing, Nothing]
```

Comment:
I'd still try removing this and keeping only `HttpError` - as by the default String-case you'd need to match both types of errors, which only one could ever occur.

This would also make `ErrorResponseAs` a total function

Reply:
Hmm I think that it is more user-friendly to expose api with partialFunction but I understand your argument. Changed.